src/Git/Value/BranchName.php

```diff
@@ -23,12 +20,14 @@ private function __construct(string $name)
         $this->name = $name;
     }
 
-    /** @psalm-pure */
+    /**
+     * @pure
+     * @psalm-suppress ImpureFunctionCall the {@see \Psl\Type\non_empty_string()} API is pure by design
```

Comment:
Is this function not declared pure? It should be :thinking:

Reply:
yea, i figure i will do it in 1.8, but assert() and coerce will always be non-pure ( conditional purity )